webapp/admin_handler.go

```diff
@@ -20,8 +23,48 @@ func adminUploadHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func showAdminUploadForm(a receiver.AppEngineAPI, w http.ResponseWriter, r *http.Request) {
+	assertLoginAndRenderTemplate(a, w, r, "/admin/results/upload", "admin_upload.html", nil)
+}
+
+func adminFlagsHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := shared.NewAppEngineContext(r)
+	a := receiver.NewAppEngineAPI(ctx)
+
+	data := struct {
+		Host string
+	}{
+		Host: shared.GetHostname(a.Context()),
+	}
+	if r.Method == "GET" {
+		assertLoginAndRenderTemplate(a, w, r, "/admin/flags", "admin_flags.html", data)
+	} else if r.Method == "POST" {
+		if !a.IsAdmin() {
+			http.Error(w, "Admin only", http.StatusUnauthorized)
+			return
+		}
+		var flag shared.Flag
+		if bytes, err := ioutil.ReadAll(r.Body); err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		} else if err = json.Unmarshal(bytes, &flag); err != nil {
```

Comment:
Nit: all these if blocks have early return. I'd personally prefer getting rid of the redundant `else`s.

Reply:
So `bytes` won't exist outside the if unless I declare it first, which means it's like, an extra 4 lines to do it that way.